sandy/core/endf6.py

```diff
@@ -2475,3 +2475,162 @@ def covariance_info(self, nubar=True, xs=True, mubar=True, chi=True):
+                        foo = sandy.EnergyCov.from_lb6
+                        cov = foo(nisec["EK"], nisec["EL"], nisec["FKL"])
+                    else:
+                        logging.warning(f"skip 'LB={lb}' covariance for"
+                                        f" [({mat}/{mt}), ({mat1}/{mt1})]")
+                        continue
+                    covs.append(cov)
+                if not covs:
+                    continue
+                cov = sandy.EnergyCov.sum_covs(*covs)
+                idx = pd.MultiIndex.from_product(
+                    [[mat], [mt], list(cov.data.index)],
+                    names=["MAT", "MT", "E"],
+                    )
+                idx1 = pd.MultiIndex.from_product(
+                    [[mat], [mt1], list(cov.data.columns)],
+                    names=["MAT1", "MT1", "E1"],
+                    )
+                df = pd.DataFrame(cov.data.values, index=idx, columns=idx1) \
+                       .stack(level=["MAT1", "MT1", "E1"]) \
+                       .rename("VAL") \
+                       .reset_index()
+                data.append(df)
+        if not data:
+            return None
+        df = pd.concat(data)
+        return sandy.CategoryCov.from_stack(df, index=["MAT", "MT", "E"],
+                                            columns=["MAT1", "MT1", "E1"],
+                                            values='VAL')
+
+    def _get_lpc_cov(self):
+        return
+
+    def _get_edistr_cov(self):
+        return
+
+    def get_cov(self, process_mf=[], mf=None, **kwds_njoy):
+        """
+        Get covariance matrices for nubar/xs/energy distribution/Legendre
+        polynomial coefficients from Endf6 or process with NJOY.
+
+        Parameters
+        ----------
+        process_mf : `int` or `list`, optional
+            DESCRIPTION. The default is [].
+        mf : `int` or `list`, optional
+            DESCRIPTION. The default is None.
+        **kwds_njoy : `dict`
+            Extra arguments for proccessing the covariance matrix with NJOY.
+
+        Returns
+        -------
+        cov : `dict` or `sandy.CategoryCov`
+            Dictionary containing the covariance matrix of different mf data.
+            If only one mf is introduced, the method returns the
+            `sandy.CategoryCov` object directly.
+
+        Examples
+        --------
+        Nubar covariance matrix from endf6 file:
+        >>> endf6 = sandy.get_endf6_file('jeff_33','xs', 922380)
+        >>> xs_cov = sandy.XsCov.from_endf6(endf6)
+        >>> nubar_cov = endf6.get_cov(mf=31)
+        >>> assert (nubar_cov.data.values == xs_cov.loc[(9237, 456), (9237, 456)].values).all().all()
+
+        Xs covariance matrix from endf6 file:
+        >>> endf_xs_cov = endf6.get_cov(mf=33)
+        >>> size = len(endf_xs_cov.data)
+        >>> assert (endf_xs_cov.data.values == xs_cov.iloc[:size, :size].values).all().all()
+
+        Both together:
+        >>> endf_xs_cov = endf6.get_cov(mf=[31, 33])
```

Comment:
This tests imply that the default is to read mf31 and mf33 from the endf6 file, which shouldn't be